lib/ecto/multi.ex

```diff
@@ -274,17 +274,17 @@ defmodule Ecto.Multi do
   The function should return either `{:ok, value}` or `{:error, value}`, and
   receives changes so far as an argument.
   """
-  @spec run(t, name, run) :: t
-  def run(multi, name, run) when is_function(run, 1) do
-    add_operation(multi, name, {:run, run})
+  @spec run(t, name, fun) :: t
```

Comment:
If we update the spec to `(t -> {:ok | :error, any})` we should also update the docs to not say: " function should return either `{:ok, value}` or `{:error, value}`".